### PR TITLE
refactor: safe exec commands

### PR DESCRIPTION
**Pull request by TrafalgarZZZ**
### Ⅰ. Describe what this PR does
- Rename `SimpleCommand` to `Command`
- Augmented `PipeCommand` to `ShellCommand` which checks all `(ba)sh -c`-like commands

### Ⅱ. Does this pull request fix one issue?

NONE

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/utils/security/exec.go [hunk 1]
```diff
@@ -14,31 +14,42 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package utils
+package security
```
Comment (cheyang) [inline pkg/utils/security/exec.go:17]: How about naming `cmdguard`?

**Review by cheyang — CHANGES_REQUESTED** [review]

File: pkg/utils/security/shell_pipes.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
+/*
```
Comment (cheyang) [inline pkg/utils/security/shell_pipes.go:1]: How about naming it `exec_pipes.go`?

File: pkg/utils/helm/helm.go [hunk 3]
```diff
@@ -70,12 +71,13 @@ func GetChartVersion(chart string) (version string, err error) {
...
 	// 2. prepare the arguments
-	args := []string{binary, "inspect", "chart", chart,
+	args := []string{helmCmd[0], "inspect", "chart", chart,
 		"|", "grep", "version:"}
 	log.V(1).Info("Exec bash -c", "args", args)
 
 	// cmd := exec.Command("bash", "-c", strings.Join(args, " "))
-	cmd, err := utils.PipeCommand("bash", "-c", strings.Join(args, " "))
+	// TODO: This should be simplified to a no-bash version
+	cmd, err := security.ShellCommand("bash", "-c", strings.Join(args, " "))
```
Comment (cheyang) [inline pkg/utils/helm/helm.go:80]: How about using an unified entry like `cmdguard.Command` for pipeCommand and simpleCommand?

Comment (TrafalgarZZZ) [reply pkg/utils/helm/helm.go]: Turned out that this is in a unused function. Already removed it.

File: pkg/utils/security/shell_pipes.go (hunk 2 above)
Comment (TrafalgarZZZ) [reply pkg/utils/security/shell_pipes.go]: Done

File: pkg/utils/security/exec.go (hunk 1 above)
Comment (TrafalgarZZZ) [reply pkg/utils/security/exec.go]: Done.

File: pkg/utils/cmdguard/exec_pipes.go [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+	"grep":  PrefixMatch,
+	"wc -l": ExactMatch, // means exact match (wc -l is exactly the allowed command)
+	// Add more commands as you see fit
+}
+
+// Define illegal sequences that may lead to command injection attack
+var illegalSequences = []string{"&", ";", "$", "'", "`", "(", ")", "||", ">>"}
+
+var allowedEnvs = []string{
+	"${METAURL}", // JuiceFS community's metaurl
```
Comment (cheyang) [inline pkg/utils/cmdguard/exec_pipes.go:62]: How about using allowedExpressions to replace allowedEnvs?

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**